sysl2/sysl/listener_impl.go

```diff
@@ -290,76 +289,58 @@ func (s *TreeShapeListener) EnterTypes(ctx *parser.TypesContext) {
 }
 
 // ExitTypes is called when production types is exited.
-func (s *TreeShapeListener) ExitTypes(ctx *parser.TypesContext) {}
+func (s *TreeShapeListener) ExitTypes(*parser.TypesContext) {}
 
 // EnterSet_type is called when production set_type is entered.
-func (s *TreeShapeListener) EnterSet_type(ctx *parser.Set_typeContext) {}
+func (s *TreeShapeListener) EnterSet_type(*parser.Set_typeContext) {}
 
-// ExitSet_type is called when production set_type is exited.
-func (s *TreeShapeListener) ExitSet_type(ctx *parser.Set_typeContext) {
-	type1 := s.typemap[s.fieldname[len(s.fieldname)-1]]
-
-	s.typemap[s.fieldname[len(s.fieldname)-1]] = &sysl.Type{
-		Type: &sysl.Type_Set{
-			Set: type1,
-		},
+// exitSetOrSequence_type is common between ExitSet_type and ExitSequence_type.
+func (s *TreeShapeListener) exitSetOrSequence_type(sizeSpec parser.ISize_specContext) (type1, newType1 *sysl.Type) {
+	type1 = s.currentType()
+	newType1 = &sysl.Type{
 		SourceContext: type1.SourceContext,
 		Opt:           type1.Opt,
 		Attrs:         type1.Attrs,
 	}
+	s.setCurrentType(newType1)
 
 	type1.Opt = false
 
 	if type1.Attrs != nil {
 		type1.Attrs = nil
 		s.popScope()
-		s.pushScope(s.typemap[s.fieldname[len(s.fieldname)-1]])
+		s.pushScope(newType1)
 	}
 
-	if ctx.Size_spec() != nil {
+	if sizeSpec != nil {
 		if type1.GetPrimitive() != sysl.Type_NO_Primitive {
-			spec := ctx.Size_spec().(*parser.Size_specContext)
+			spec := sizeSpec.(*parser.Size_specContext)
 			type1.Constraint = makeTypeConstraint(type1.GetPrimitive(), spec)
 		}
 	}
+	return
```

Comment:
hmm learned a new thing today.

Reply:
Yeah, empty returns at the end of a function are not my cup of tea.